### PR TITLE
add objectives and tasks for Nix team

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
so far there were no even remotely measurable objectives, only a general
purpose statement.

this change is intended to focus the team's work on what I (and many
others I talked to) perceive to be the main pain point in the
development process.

This work is sponsored by [Antithesis](https://antithesis.com/) ✨

### CODE REVIEW

**Review by rapenne-s — APPROVED** [review]

File: maintainers/README.md [hunk 1]
```diff
@@ -2,7 +2,22 @@
...
+- It is clear what is expected of a pull request.
+- It is predictable how to get a change merged and released.
+
+### Tasks
+
+- Elaborate contribution guides and abide to them
+- Improve documentation targeted to contributors
+- Maintain the issue tracker and triage pull requests
+- Help competent contributors succeed with their pull requests
+- Ensure the required maintainer capacity for all of the above
```
Comment (thufschmitt) [inline maintainers/README.md:20]: Should consider the responsibilities from [The team creation's post](https://discourse.nixos.org/t/nix-team-creation/22228)

File: maintainers/README.md [hunk 2]
```diff
@@ -2,7 +2,22 @@
...
 
-The goal of the team is to help other people to contribute to Nix.
+In order to attract more maintainers and thereby scale the development process to match user needs, the main goal of this team is to improve the Nix contributor experience.
+
+### Objectives
+
+- It is obvious what is worthwhile to work on.
+- It is easy to find the right place in the code to make a change.
+- It is clear what is expected of a pull request.
+- It is predictable how to get a change merged and released.
```
Comment (thufschmitt) [inline maintainers/README.md:12]: I think this is missing a few important items: 
- The team should take responsibility for making sure that the codebase stays in a good shape (beyond just “enabling other people to do the job”). In particular, that means handling a few key aspects of the development like:
    - Release management
    - Making sure that _someone_ is available to fix the most critical bugs
- There's an “external communication” aspect that we haven't been keeping up too much but is important (telling the world how great contributing to and using Nix is)
- The team should also set the expectations for users (roadmap & regular releases in particular)

**Review by thufschmitt — CHANGES_REQUESTED** [review]

**Comment by fricklerhandwerk** [comment]
Discussed in Nix team meeting 2023-01-06:

- @thufschmitt: this draft differs quite substantially from the [original team announcement](https://discourse.nixos.org/t/nix-team-creation/22228)
- @Ericson2314, @tomberek: as long as it does not deviate from our general course we should merge it and improve later as needed
- @edolstra: enabling people to contribute is not the stated primary goal of the team. right now the people on the team do 95% of the work
    - making contributions easier is only part of our responsibilities
    - @Ericson2314: agree, but we should still highlight this more
    - @thufschmitt: agree, end goal is not to help others to do the work, the end goal is to make Nix great
    - also this is not how most open source software projects work. maintainers do almost everything
        - @fricklerhandwerk: that's true and it's that way for a reason. there are certain types of work volunteers are highly probable to do (cool deeply technical things and new features). and then there are some that no one likes to do, not even maintainers. see the [Roads and Bridges report](https://www.fordfoundation.org/work/learning/research-reports/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure/) on maintaining open source projects sustainibly ([found here](https://discourse.nixos.org/t/my-nixpkgs-trust-worries/24213/22))
            - @edolstra: right, building new things is more pleasant than reviewing pull requests
            - @fricklerhandwerk: the maintainer team has the capacity to do the dirty work. in the extreme, we should not even implement anything at all and completely focus on guiding other people's efforts such that they fit into the big picture
    - we have other responsibilities, such as creating a roadmap or making releases
        - @fricklerhandwerk: yes, and one can view those as a support function to the general contribution cycle
- @fricklerhandwerk: would like to establish that contributing should become a first-class use case
    - no agreement
- @fricklerhandwerk: will reword the proposal based on feedback. will make more explicit the separation of objectives, results, and tasks/responsibilities and soften the strong stance on enabling contributors
    - agreement
- @edolstra: meta-point: team description should be in sync with nixos.org
    - also teams should be on its own page
    - agreement

File: maintainers/README.md (hunk 2 above)
Comment (blaggacao) [reply maintainers/README.md:12]: - Architecture & Design!

**Comment by blaggacao** [comment]
If not yet on the radar, I candidly request the Team to engage more in "Architecture & Design" work to give the high level directional guidance and shorten any feedback loop on yhe big picture.

That's likely the single most profitable investment that currently could be made on behalf of the project.

File: maintainers/README.md (hunk 1 above)
Comment (fricklerhandwerk) [reply maintainers/README.md]: Done

File: maintainers/README.md (hunk 2 above)
Comment (fricklerhandwerk) [reply maintainers/README.md:14]: More clearly separated objectives from tasks.

**Comment by Ericson2314** [comment]
We reviewed this in the Nix team meeting today

@roberth proposed that we should be explicit that writing code is a non-objective: as developers we act outside of the team, and so there is a "level playing field" for non-team-member and team-member developers alike.

The team agreed, but that this can be added to this document right after in a separate follow-up PR. (Thus, the PR was merged.) @roberth will make that PR.

**Pull request merged**